airflow/executors/kubernetes_executor.py

```diff
@@ -895,11 +895,11 @@ def _change_state(self,
                       pod_id: str,
                       namespace: str) -> None:
         if state != State.RUNNING:
-            if self.kube_config.delete_worker_pods_on_success and state is State.SUCCESS:
+            if self.kube_config.delete_worker_pods and state is State.SUCCESS:
                 if not self.kube_scheduler:
                     raise AirflowException(NOT_STARTED_MESSAGE)
                 self.kube_scheduler.delete_pod(pod_id, namespace)
-            elif self.kube_config.delete_worker_pods:
+            elif self.kube_config.delete_worker_pods_on_failure:
```

Comment:
How about changing the semantics a bit here: if we make `delete_worker_pods` be the "enable/disable deletion at all" and this new setting only controls what happens to failed pods:

```python
            if self.kube_config.delete_worker_pods:
               if not self.kube_scheduler:
                   raise AirflowException(NOT_STARTED_MESSAGE)
               if state is not State.FAILURE or self.delete_worker_pods_on_failure
                   self.kube_scheduler.delete_pod(pod_id, namespace)
```

And then the description of the config becomes:

```ini
# If True, worker pods will be deleted upon termination
delete_worker_pods = True

# If False (and delete_worker_pods is True), failed worker pods will not be deleted so users can investigate them.
delete_worker_pods_on_failure = True
```

Reply:
Sure, LGTM